### PR TITLE
Add GetInstalledPackageDetail implementation.

### PR DESCRIPTION
**Pull request by absoludity**
### Description of the change

Adds implementation for GetInstalledPackageDetail.

The response includes the release values, notes and a ref to the available package for further details about the available package:
```
# Get the summaries first to find an installed package ref:
$ grpcurl -plaintext localhost:8080 kubeappsapis.plugins.helm.packages.v1alpha1.HelmPackagesService.GetInstalledPackageSummaries                                                                                               {
  "installedPackageSummaries": [
    ...
    {
      "installedPackageRef": {
        "context": {
          "namespace": "default"
        },
        "identifier": "test-apache"
      },
      "name": "test-apache",
      "pkgVersionReference": {
        "version": "8.5.8"
      },
      "currentPkgVersion": "8.5.8",
      "currentAppVersion": "2.4.48",
      "iconUrl": "https://bitnami.com/assets/stacks/apache/img/apache-stack-220x234.png",
      "pkgDisplayName": "apache",
      "shortDescription": "Chart for Apache HTTP Server",
      "latestPkgVersion": "8.6.0",
      "status": {
        "ready": true,
        "reason": "STATUS_REASON_INSTALLED",
        "userReason": "deployed"
      }
    }
  ]
}

# Now grab the detail:
$ grpcurl -plaintext -d '{"installed_package_ref": {"context": {"namespace": "default"}, "identifier": "test-apache"}}' localhost:8080  kubeappsapis.plugins.helm.packages.v1alpha1.HelmPackagesService.GetInstalledPackageDetail
{
  "installedPackageDetail": {
    "installedPackageRef": {
      "context": {
        "namespace": "default"
      },
      "identifier": "test-apache"
    },
    "pkgVersionReference": {
      "version": "8.5.8"
    },
    "name": "test-apache",
    "currentPkgVersion": "8.5.8",
    "valuesApplied": "{\"affinity\":{},\"cloneHtdocsFromGit\":{\"enabled\":false,\"extraVolumeMounts\":[],\"interval\":60,\"resources\":{}},\"commonAnnotations\":{},\"commonLabels\":{},\"extraDeploy\":[],\"extraEnvVars\":[],\"extraVolumeMounts\":[],\"extraVolumes\":[],\"git\":{\"pullPolicy\":\"IfNotPresent\",\"registry\":\"docker.io\",\"repository\":\"bitnami/git\",\"tag\":\"2.32.0-debian-10-r26\"},\"hostAliases\":[{\"hostnames\":[\"status.localhost\"],\"ip\":\"127.0.0.1\"}],\"image\":{\"debug\":false,\"pullPolicy\":\"IfNotPresent\",\"registry\":\"docker.io\",\"repository\":\"bitnami/apache\",\"tag\":\"2.4.48-debian-10-r28\"},\"ingress\":{\"annotations\":{},\"apiVersion\":null,\"certManager\":false,\"enabled\":false,\"hostname\":\"example.local\",\"path\":\"/\",\"pathType\":\"ImplementationSpecific\",\"secrets\":null,\"tls\":[{\"hosts\":[\"example.local\"],\"secretName\":\"example.local-tls\"}]},\"initContainers\":{},\"kubeVersion\":null,\"livenessProbe\":{\"enabled\":true,\"failureThreshold\":6,\"initialDelaySeconds\":180,\"path\":\"/\",\"periodSeconds\":20,\"port\":\"http\",\"successThreshold\":1,\"timeoutSeconds\":5},\"metrics\":{\"enabled\":false,\"image\":{\"pullPolicy\":\"IfNotPresent\",\"registry\":\"docker.io\",\"repository\":\"bitnami/apache-exporter\",\"tag\":\"0.9.0-debian-10-r25\"},\"podAnnotations\":{\"prometheus.io/port\":\"9117\",\"prometheus.io/scrape\":\"true\"},\"resources\":{\"limits\":{},\"requests\":{}}},\"nodeAffinityPreset\":{\"key\":\"\",\"type\":\"\",\"values\":[]},\"nodeSelector\":{},\"podAffinityPreset\":\"\",\"podAnnotations\":{},\"podAntiAffinityPreset\":\"soft\",\"podLabels\":{},\"readinessProbe\":{\"enabled\":true,\"failureThreshold\":6,\"initialDelaySeconds\":30,\"path\":\"/\",\"periodSeconds\":10,\"port\":\"http\",\"successThreshold\":1,\"timeoutSeconds\":5},\"replicaCount\":1,\"resources\":{\"limits\":{},\"requests\":{}},\"service\":{\"annotations\":{},\"externalTrafficPolicy\":\"Cluster\",\"httpsPort\":443,\"nodePorts\":{\"http\":\"\",\"https\":\"\"},\"port\":80,\"type\":\"LoadBalancer\"},\"sidecars\":{},\"tolerations\":[]}",
    "status": {
      "ready": true,
      "reason": "STATUS_REASON_INSTALLED",
      "userReason": "deployed"
    },
    "postInstallationNotes": "\n** Please be patient while the chart is being deployed **\n\n1. Get the Apache URL by running:\n\n** Please ensure an external IP is associated to the test-apache service before proceeding **\n** Watch the status using: kubectl get svc --namespace default -w test-apache **\n\n  export SERVICE_IP=$(kubectl get svc --namespace default test-apache --template \"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}\")\n  echo URL            : http://$SERVICE_IP/\n\n\nWARNING: You did not provide a custom web application. Apache will be deployed with a default page. Check the README section \"Deploying your custom web application\" in https://github.com/bitnami/charts/blob/master/bitnami/apache/README.md#deploying-your-custom-web-application.\n\n\n",
    "availablePackageRef": {
      "context": {
        "namespace": "kubeapps"
      },
      "identifier": "bitnami/apache",
      "plugin": {
        "name": "helm.packages",
        "version": "v1alpha1"
      }
    }
  }
}

```

We'll need to enable querying the k8s resources for an installed package separately, rather than including those details in this response. So the related UX calls will still use the old API until we do so.

### Benefits

One step closer.

### Applicable issues

  - Ref #3146

### CODE REVIEW

File: script/deploy-dev.mk [hunk 1]
```diff
@@ -37,6 +37,17 @@ deploy-dev-kubeapps:
 		--values ./docs/user/manifests/kubeapps-local-dev-auth-proxy-values.yaml \
 		--values ./docs/user/manifests/kubeapps-local-dev-additional-kind-cluster.yaml
 
+deploy-dev-kubeapps-with-apis:
```
Comment (absoludity) [inline script/deploy-dev.mk:40]: Not sure if others would find this useful until we have support in the chart, but I find it really handy to build the image tag, load it to my kind cluster, then just do `make deploy-dev-kubeapps-with-apis` . Obviously, you need to update the image tag, and if you want redis enabled etc.

Comment (antgamdia) [reply script/deploy-dev.mk:40]: Great, thank you! I ended up creating a separate local values.yaml file and a local script doing: 

```bash
kubectl scale deploy/kubeapps-internal-kubeappsapis -n kubeapps --replicas=0
IMAGE_TAG=dev1 make kubeapps/kubeapps-apis
kind load docker-image kubeapps/kubeapps-apis:dev1 --name kubeapps-for-pinniped
kubectl set image deployment/kubeapps-internal-kubeappsapis -n kubeapps kubeappsapis=kubeapps/kubeapps-apis:dev1 --record
kubectl scale deploy/kubeapps-internal-kubeappsapis -n kubeapps --replicas=1
```
But having it there is neater!  Thanks!

File: cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go [hunk 2]
```diff
@@ -648,3 +650,80 @@ func installedPkgSummaryFromRelease(r *release.Release) *corev1.InstalledPackage
...
+		Namespace:  release.Namespace,
+		ChartName:  release.Chart.Metadata.Name,
+		Version:    release.Chart.Metadata.Version,
+		AppVersion: release.Chart.Metadata.AppVersion,
+	}
+	charts, _, err := s.manager.GetPaginatedChartListWithFilters(cq, 1, 0)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Error while fetching related chart: %v", err)
+	}
+	if len(charts) == 1 {
```
Comment (antgamdia) [inline cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go:698]: If len>1 shouldn't it return an error/warning ?

**Review by antgamdia — APPROVED** [review]
Awesome! Once we get it landed, we will have almost every required operations for building the UI :)   Thanks!

**Comment by antgamdia** [comment]
Let me merge this PR for starting working on the required UI changes :)

**Pull request merged**